Extensions/TweenBehavior/tweenruntimebehavior.ts

```diff
@@ -632,39 +654,192 @@ namespace gdjs {
+      let fromColor: any[] = this.owner.getColor().split(';');
+      if (useHSLColorTransition) {
+        fromColorAsHSL = rgbToHsl(fromColor);
+      }
+      let toColor: any[] = toColorStr.split(';');
       if (toColor.length !== 3) {
         return;
       }
+
       const newTweenable = TweenRuntimeBehavior.makeNewTweenable(
         this._runtimeScene
       );
+
+      if(useHSLColorTransition) {
+        toColorAsHSL = rgbToHsl(toColor);
+
+        newTweenable.setConfig({
+          from: { hue: fromColorAsHSL[0], saturation: fromColorAsHSL[1], lightness: fromColorAsHSL[2] },
+          to: { hue: toColorAsHSL[0], saturation: toColorAsHSL[1], lightness: toColorAsHSL[2] },
+          duration: durationValue,
+          easing: easingValue,
+          step: function step(state) {
+            const rgbFromHslColor = hslToRgb([state.hue, state.saturation, state.lightness]);
+            // @ts-ignore - objects are duck typed
+            that.owner.setColor(
+              Math.floor(rgbFromHslColor[0]) +
+                ';' +
+              Math.floor(rgbFromHslColor[1]) +
+                ';' +
+              Math.floor(rgbFromHslColor[2])
+            );
+          },
+        });
+      } else {
+        newTweenable.setConfig({
+          from: { red: fromColor[0], green: fromColor[1], blue: fromColor[2] },
+          to: { red: toColor[0], green: toColor[1], blue: toColor[2] },
+          duration: durationValue,
+          easing: easingValue,
+          step: function step(state) {
+            // @ts-ignore - objects are duck typed
+            that.owner.setColor(
+              Math.floor(state.red) +
+                ';' +
+                Math.floor(state.green) +
+                ';' +
+                Math.floor(state.blue)
+            );
+          },
+        });
+      }
+      this._addTween(
+        identifier,
+        newTweenable,
+        this._runtimeScene.getTimeManager().getTimeFromStart(),
+        durationValue
+      );
+      this._setupTweenEnding(identifier, destroyObjectWhenFinished);
+    }
+
+
+
+    /**
+     * Add an object color HSL tween, with the "to" color given using HSL (H: any number, S and L: 0-100).
+     * @param identifier Unique id to idenfify the tween
+     * @param toHue The target hue, or the same as the from color's hue if blank
+     * @param animateHue, include hue in calculations, as can't set this to -1 as default to ignore
+     * @param toSaturation The target saturation, or the same as the from color's saturation if blank
+     * @param toHue The target lightness, or the same as the from color's lightness if blank
+     * @param easingValue Type of easing
+     * @param durationValue Duration in milliseconds
+     * @param destroyObjectWhenFinished Destroy this object when the tween ends
+     */
+    
+    addObjectColorHSLTween(
+      identifier: string,
+      toHue: string,
+      animateHue: string,
+      toSaturation: string,
```

Comment:
These are numbers, not string.

Reply:
They are (and a boolean), and I've tweaked the code to reflect this.